schema/fields_test.go

```diff
@@ -101,4 +101,59 @@ func TestFieldBuilder_Build(t *testing.T) {
 		_, err := builder.Build()
 		require.Error(t, err)
 	})
+
+	t.Run("test programming language keywords as field name", func(t *testing.T) {
+		builder1 := &FieldBuilder{FieldName: "abstract", Type: "string"}
+		_, err1 := builder1.Build()
+		require.Equal(t, err1, api.Errorf(api.Code_INVALID_ARGUMENT, "Invalid collection field name, It contains language keyword for fieldName = 'abstract'"))
+
+		builder2 := &FieldBuilder{FieldName: "integer", Type: "string"}
+		_, err2 := builder2.Build()
+		require.Equal(t, err2, api.Errorf(api.Code_INVALID_ARGUMENT, "Invalid collection field name, It contains language keyword for fieldName = 'integer'"))
+
+		builder3 := &FieldBuilder{FieldName: "yield", Type: "string"}
+		_, err3 := builder3.Build()
+		require.Equal(t, err3, api.Errorf(api.Code_INVALID_ARGUMENT, "Invalid collection field name, It contains language keyword for fieldName = 'yield'"))
+	})
+
+	t.Run("test invalid field name pattern", func(t *testing.T) {
+		builder1 := &FieldBuilder{FieldName: "0id", Type: "string"}
+		_, err1 := builder1.Build()
+		require.Equal(t, err1, api.Errorf(api.Code_INVALID_ARGUMENT, "Invalid collection field name, field name can only contain [a-zA-Z0-9_$] and it can only start with [a-zA-Z_$] for fieldName = '0id'"))
```

Comment:
What are your thoughts on incorporate `MsgFieldNameInvalidPattern` in the second argument of `api.Errorf()`?

Reply:
updated.